### PR TITLE
[receiver/kafkareceiver]: allow tunable fetch sizes

### LINKED ISSUE #22741: How to optimize Kafka receiver for slow consumption of messages?
### Component(s)

receiver/kafka

### Describe the issue you're reporting

Kafka receiver is only consuming a few dozen messages per second, how can I optimize it?

```
receivers:
  kafka/1:
    brokers:
    - 10.xx.xx.x1:9092
    - 10.xx.xx.x2:9092
    - 10.xx.xx.x3:9092
    protocol_version: 2.0.0
    topic: otlp-spans-topic-01
    group_id: otel-collector-01
    encoding: jaeger_json      
  kafka/2:
    brokers:
     - 10.xx.xx.x1:9092
     - 10.xx.xx.x2:9092
     - 10.xx.xx.x3:9092
    protocol_version: 2.0.0
    topic: otlp-spans-topic-02
    group_id: otel-collector-02
    encoding: jaeger_json  

service:
    traces:
      exporters:
      - clickhousetraces
      processors:
      - signozspanmetrics/prometheus
      - batch
      receivers:
      - otlp
      - kafka/1
      - kafka/2
      - jaeger 
 
 ```

### PR DESCRIPTION
**Pull request by cxdy**
**Description:**
This commit adds the ability to tune the minimum, default and maximum fetch sizes for the Kafka Receiver in the OpenTelemetry configuration file.

The defaults are kept consistent with the defaults imposed by [sarama](https://pkg.go.dev/github.com/shopify/sarama#Config)

**Link to tracking Issue:** 
Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22741

**Testing:**
Built the image with `make docker-otelcontribcol`, uploaded to [Docker Hub](https://hub.docker.com/r/ckaczynski/otelcontribcol) to pull the image into our Kubernetes deployment, and hit the cluster with 200k logs per second with the following settings:
```
        default_fetch_size: 15728640 # 15MB
        max_fetch_size: 31457280 # 30MB
        min_fetch_size: 1048576  # 1MB
```
These limits are absurdly high for no reason other than because I just wanted to set them higher and see how far I could push it in our cluster.

No new tests were added, but added to existing configuration tests in config.go and factory.go

**Documentation:** 
Added the new configuration options to the receiver's README

### CODE REVIEW

**Comment by cxdy** [comment]
Sorry about the merge commits, can probably squash them at merge but I can rebase too.

**Comment by cxdy** [comment]
Ah, apologies, missed the changelog entry. Will add shortly.

**Comment by cxdy** [comment]
Unit tests _should_ pass now, I didn't realize I needed to specify `int32` for the type in `factory_test.go`.

**Review by mwear — APPROVED** [review]
LGTM. Could the codeowners take a look cc: @pavolloffay @MovieStoreGuy.

File: receiver/kafkareceiver/config.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ type Config struct {
 
 	// Extract headers from kafka records
 	HeaderExtraction HeaderExtraction `mapstructure:"header_extraction"`
+
+	// The minimum bytes per fetch from Kafka (default "1")
+	MinFetchSize int32 `mapstructure:"min_fetch_size"`
+	// The default bytes per fetch from Kafka (default "1048576")
+	DefaultFetchSize int32 `mapstructure:"default_fetch_size"`
+	// The maximum bytes per fetch from Kafka (default "0")
```
Comment (iblancasa) [inline receiver/kafkareceiver/config.go:86]:
```suggestion
	// The maximum bytes per fetch from Kafka (default "0", no limit)
```

**Review by iblancasa — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Can I ask you to fix up the conflicts?

**Comment by cxdy** [comment]
Sure can!

File: receiver/kafkareceiver/factory_test.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ func TestCreateDefaultConfig(t *testing.T) {
 	assert.Equal(t, defaultInitialOffset, cfg.InitialOffset)
 	assert.Equal(t, defaultSessionTimeout, cfg.SessionTimeout)
 	assert.Equal(t, defaultHeartbeatInterval, cfg.HeartbeatInterval)
+	assert.Equal(t, int32(defaultMinFetchSize), cfg.MinFetchSize)
```
Comment (MovieStoreGuy) [inline receiver/kafkareceiver/factory_test.go:31]: ``suggestion
	assert.Equal(t, defaultMinFetchSize, cfg.MinFetchSize)
	assert.Equal(t, defaultDefaultFetchSize, cfg.DefaultFetchSize)
	assert.Equal(t, defaultMaxFetchSize, cfg.MaxFetchSize)
```

To fix up the unneeded conversions.

File: receiver/kafkareceiver/factory_test.go [hunk 3]
```diff
@@ -28,6 +28,9 @@ func TestCreateDefaultConfig(t *testing.T) {
...
+	assert.Equal(t, int32(defaultDefaultFetchSize), cfg.DefaultFetchSize)
+	assert.Equal(t, int32(defaultMaxFetchSize), cfg.MaxFetchSize)
```
Comment (MovieStoreGuy) [inline receiver/kafkareceiver/factory_test.go:33]:
```suggestion
	assert.Equal(t, defaultMinFetchSize, cfg.MinFetchSize)
	assert.Equal(t, defaultDefaultFetchSize, cfg.DefaultFetchSize)
	assert.Equal(t, defaultMaxFetchSize, cfg.MaxFetchSize)
```

To fix up the unneeded conversions.

**Review by MovieStoreGuy — APPROVED** [review]
Just need to fix up conversions.

**Comment by cxdy** [comment]
Thanks for that, haven't had a chance to grab my laptop yet. Committed!

**Pull request merged**